src/de/map.rs

```diff
@@ -163,11 +163,11 @@ impl<'de, 'a> de::Deserializer<'de> for MapKey<'a, 'de> {
         self.de.deserialize_str(visitor)
     }
 
-    fn deserialize_string<V>(self, _visitor: V) -> Result<V::Value, Self::Error>
+    fn deserialize_string<V>(self, visitor: V) -> Result<V::Value, Self::Error>
     where
         V: Visitor<'de>,
     {
-        unreachable!()
```

Comment:
I don't know why `deserialize_string()` is called here as it is [defined as unreachable](https://github.com/rust-embedded-community/serde-json-core/blob/master/src/de/map.rs#L169) in the embedded version.

Reply:
I'm not entirely sure about the details and don't fully understand the `Deserializer` trait. The difference has something to do with deserializing into borrowed `str`s and owned `String`s. So it might be caused by your `balances: BTreeMap<String,u16>,` test. The work in https://github.com/rust-embedded-community/serde-json-core/pull/23 is lacking tests, so it may be broken but undetected.

Regarding the implementation: I think it's better to forward both `deserialize_str` and `deserialize_string` to `de` and let it do something useful with it. I.e. do:

```rust
self.de.deserialize_string(visitor)
```